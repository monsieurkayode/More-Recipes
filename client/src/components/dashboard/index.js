/* jshint esversion: 6 */
import UserProfile from './UserProfile';
import UserRecipe from './UserRecipe';
import WelcomeDisplay from './WelcomeDisplay';
import DashboardPanel from './DashboardPanel';
import SideNavDashboard from './SideNavDashboard';
import UserFavoriteRecipe from './UserFavoriteRecipe';
import UserNotification from './UserNotification';

export {
  UserProfile,
  UserRecipe,
  UserFavoriteRecipe,
  UserNotification,
  WelcomeDisplay,
  SideNavDashboard,
  DashboardPanel
};
