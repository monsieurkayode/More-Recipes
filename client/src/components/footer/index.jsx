import React from 'react';

/**
 * Footer
 * @function Footer
 *
 * @param {void} void
 *
 * @return {JSX} JSX
 */
const Footer = () => (
  <footer className="page-footer blue-grey darken-2">
    <div className="row">
      <div className="col l3 social-icon-container">
        <ul>
          <li>
            <div>
              <span>
                <i className="fa fa-2x fa-github teal-text" />
              </span>
            </div>
          </li>
          <li>
            <div>
              <span>
                <i className="fa fa-2x fa-facebook teal-text" />
              </span>
            </div>
          </li>
          <li>
            <div>
              <span>
                <i className="fa fa-2x fa-twitter teal-text" />
              </span>
            </div>
          </li>
          <li>
            <div>
              <span>
                <i className="fa fa-2x fa-google teal-text" />
              </span>
            </div>
          </li>
        </ul>
      </div>
    </div>
    <div className="footer-copyright">
      <div className="container">
        Â© Andela 2018
        <a
          className="grey-text text-lighten-4 right"
          href="/"
        >
        Created by Kayode Adeola
        </a>
      </div>
    </div>
  </footer>
);

export default Footer;
