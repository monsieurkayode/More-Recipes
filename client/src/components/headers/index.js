/* jshint esversion: 6 */
import Banner from './Banner';
import DashboardNavbar from './DashboardNavbar';
import HomeNavbar from './HomeNavbar';
import LandingNavbar from './LandingNavbar';
import UserMenu from './UserMenu';
import SearchBar from './SearchBar';

export {
  Banner,
  DashboardNavbar,
  HomeNavbar,
  LandingNavbar,
  UserMenu,
  SearchBar
};
