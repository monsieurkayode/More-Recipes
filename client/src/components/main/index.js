/* jshint esversion: 6 */
import Category from './Category';
import Recipes from './Recipes';
import RecipeItem from './RecipeItem';
import TopRecipeItem from './TopRecipeItem';
import TopRecipes from './TopRecipes';
import SignupForm from './SignupForm';
import SigninForm from './SigninForm';
import TextField from './TextField';
import PostRecipe from './PostRecipe';
import EditRecipe from './EditRecipe';
import SideNav from './SideNav';
import WelcomeMessage from './WelcomeMessage';
import Loader from './Loader';

export {
  Category,
  Recipes,
  RecipeItem,
  TopRecipeItem,
  TopRecipes,
  PostRecipe,
  EditRecipe,
  SignupForm,
  SigninForm,
  TextField,
  SideNav,
  WelcomeMessage,
  Loader
};
