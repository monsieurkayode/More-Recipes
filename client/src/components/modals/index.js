/* jshint esversion: 6 */
import DeleteModal from './DeleteModal';

export default DeleteModal;
