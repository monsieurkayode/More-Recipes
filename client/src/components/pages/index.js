/* jshint esversion: 6 */
import SigninPage from './SigninPage';
import SignupPage from './SignupPage';
import DashboardPage from './DashboardPage';
import NotFoundPage from './NotFoundPage';
import RecipeViewPage from './RecipeViewPage';
import PostRecipePage from './PostRecipePage';
import IndexPage from './IndexPage';
import EditRecipePage from './EditRecipePage';
import HomePage from './Homepage';
import CategoryPage from './CategoryPage';

export {
  SigninPage,
  SignupPage,
  DashboardPage,
  NotFoundPage,
  RecipeViewPage,
  PostRecipePage,
  IndexPage,
  EditRecipePage,
  HomePage,
  CategoryPage
};
