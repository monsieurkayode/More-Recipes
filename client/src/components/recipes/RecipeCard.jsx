import React from 'react';
import PropTypes from 'proptypes';
import { Link } from 'react-router-dom';
import { formatTime } from '../../utils/timeFormat';
/**
 * RecipeCard
 * @function RecipeCard
 *
 * @param {object} props
 *
 * @return {JSX} JSX
 */
const RecipeCard = (props) => {
  const { recipe, selected, selectRecipe } = props;
  let time = recipe.createdAt;
  time = formatTime(time, 'MMMM DD h:mm a');
  return (
    <div>
      <div className="card-image">
        <Link to={`/recipes/${recipe.id}`}>
          <img
            className="responsive-img"
            src={recipe.image}
            alt=""
          />
        </Link>
        <span className="card-title">{recipe.views} views</span>
      </div>
      <div className={`card-content ${props.size}`}>
        <span className="card-title truncate">
          <strong>{recipe.recipeName}</strong>
        </span>
        <div className="divider" />
        <Link
          to={selected === 'recipes' ? `/recipes/${recipe.id}/edit` :
            `/favorites/${recipe.id}/edit`}
          className="edit chip"
        >
          <i className="fa fa-pencil" /> Edit
        </Link><span>
          <a
            onClick={() => selectRecipe(recipe.id)}
            className="delete chip modal-trigger"
            href="#modal-delete"
          >
            <i className="fa fa-trash-o" />
            {selected === 'recipes' ? ' Delete' : ' Remove'}
          </a><br />
          <span className="right">
            <i className="material-icons tiny reaction">
              thumb_down</i>{recipe.downvote}
          </span>
          <span className="right">
            <i className="material-icons tiny reaction">
              thumb_up</i>{recipe.upvote}
          </span>
          <span><p className="created-at">{time}</p></span>
        </span>
      </div>
    </div>
  );
};

RecipeCard.defaultProps = {
  selected: ''
};

RecipeCard.propTypes = {
  size: PropTypes.string.isRequired,
  recipe: PropTypes.shape({
  }).isRequired,
  selected: PropTypes.string,
  selectRecipe: PropTypes.func.isRequired
};

export default RecipeCard;
