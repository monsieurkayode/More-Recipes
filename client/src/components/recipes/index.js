import RecipeCard from './RecipeCard';
import RecipeCardSmall from './RecipeCardSmall';

export {
  RecipeCard,
  RecipeCardSmall
};
