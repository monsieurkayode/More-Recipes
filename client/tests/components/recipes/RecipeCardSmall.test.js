/* eslint-disable no-undef */
import React from 'react';
import { BrowserRouter } from 'react-router-dom';

import { RecipeCardSmall } from '../../../src/components/recipes';
import data from '../../__mocks__/mockData';

const setup = (selected) => {
  const props = {
    recipe: data.recipeDetails,
    selectRecipe: jest.fn(),
    selected
  };
  const wrapper = mount(
    <BrowserRouter>
      <RecipeCardSmall {...props} />
    </BrowserRouter>
  );
  return {
    wrapper
  };
};

describe('The RecipeCardSmall Component', () => {
  it('should render correctly', (done) => {
    const { wrapper } = setup('recipes');
    expect(toJson(wrapper)).toMatchSnapshot();
    done();
  });
});
