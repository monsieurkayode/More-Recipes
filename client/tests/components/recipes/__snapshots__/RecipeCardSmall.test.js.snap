// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The RecipeCardSmall Component should render correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "blank",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <RecipeCardSmall
      recipe={
        Object {
          "category": "specials",
          "id": 1,
          "ingredients": "**A list of ingredients for preparing Chicken soup**
                  * 1 (3 pound) whole chicken
                  * 4 carrots, halved
                  * 4 stalks celery, halved
                  * 1 large onion, halved
                  * water to cover
                  * salt and pepper to taste
                  * 1 teaspoon chicken bouillon granules (optional)",
          "instructions": "* Put the chicken, carrots, celery and onion in a large soup
                  pot and cover with cold water. Heat and simmer, uncovered,
                  until the chicken meat falls off of the bones
                  (skim off foam every so often).
                  * Take everything out of the pot. Strain the broth. 
                  Pick the meat off of the bones and chop the carrots, celery 
                  and onion. Season the broth with salt, pepper and chicken 
                  bouillon to taste, if desired. Return the chicken, carrots, 
                  celery and onion to the pot, stir together, and serve.",
          "recipeName": "Chicken Soup",
        }
      }
      selectRecipe={[MockFunction]}
      selected="recipes"
    >
      <div
        className="col l3 m8 s12 offset-m2"
      >
        <div
          className="card hoverable views-small"
        >
          <RecipeCard
            recipe={
              Object {
                "category": "specials",
                "id": 1,
                "ingredients": "**A list of ingredients for preparing Chicken soup**
                  * 1 (3 pound) whole chicken
                  * 4 carrots, halved
                  * 4 stalks celery, halved
                  * 1 large onion, halved
                  * water to cover
                  * salt and pepper to taste
                  * 1 teaspoon chicken bouillon granules (optional)",
                "instructions": "* Put the chicken, carrots, celery and onion in a large soup
                  pot and cover with cold water. Heat and simmer, uncovered,
                  until the chicken meat falls off of the bones
                  (skim off foam every so often).
                  * Take everything out of the pot. Strain the broth. 
                  Pick the meat off of the bones and chop the carrots, celery 
                  and onion. Season the broth with salt, pepper and chicken 
                  bouillon to taste, if desired. Return the chicken, carrots, 
                  celery and onion to the pot, stir together, and serve.",
                "recipeName": "Chicken Soup",
              }
            }
            selectRecipe={[MockFunction]}
            selected="recipes"
            size="small-cards"
          >
            <div>
              <div
                className="card-image"
              >
                <Link
                  replace={false}
                  to="/recipes/1"
                >
                  <a
                    href="/recipes/1"
                    onClick={[Function]}
                  >
                    <img
                      alt=""
                      className="responsive-img"
                    />
                  </a>
                </Link>
                <span
                  className="card-title"
                >
                   views
                </span>
              </div>
              <div
                className="card-content small-cards"
              >
                <span
                  className="card-title truncate"
                >
                  <strong>
                    Chicken Soup
                  </strong>
                </span>
                <div
                  className="divider"
                />
                <Link
                  className="edit chip"
                  replace={false}
                  to="/recipes/1/edit"
                >
                  <a
                    className="edit chip"
                    href="/recipes/1/edit"
                    onClick={[Function]}
                  >
                    <i
                      className="fa fa-pencil"
                    />
                     Edit
                  </a>
                </Link>
                <span>
                  <a
                    className="delete chip modal-trigger"
                    href="#modal-delete"
                    onClick={[Function]}
                  >
                    <i
                      className="fa fa-trash-o"
                    />
                     Delete
                  </a>
                  <br />
                  <span
                    className="right"
                  >
                    <i
                      className="material-icons tiny reaction"
                    >
                      thumb_down
                    </i>
                  </span>
                  <span
                    className="right"
                  >
                    <i
                      className="material-icons tiny reaction"
                    >
                      thumb_up
                    </i>
                  </span>
                  <span>
                    <p
                      className="created-at"
                    >
                      February 12 3:12 am
                    </p>
                  </span>
                </span>
              </div>
            </div>
          </RecipeCard>
        </div>
      </div>
    </RecipeCardSmall>
  </Router>
</BrowserRouter>
`;
